client/lib/core/widgets/media_settings_widget.dart

```diff
@@ -0,0 +1,475 @@
+    _videoElement.remove();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      insetPadding: responsiveLayoutService.isMobile(context)
+          ? EdgeInsets.symmetric(horizontal: 20.0, vertical: 12.0)
+          : null,
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(
+            'Audio Input Device',
+            style: context.theme.textTheme.titleMedium,
+          ),
+          _mediaService.audioInputs.isEmpty
+              ? Padding(
+                  padding: const EdgeInsets.only(top: 16.0),
+                  child: Text(
+                    'No audio devices available. Please check permissions.',
+                    style: context.theme.textTheme.bodyMedium!.copyWith(
+                      color: context.theme.colorScheme.onSurfaceVariant,
+                    ),
+                  ),
+                )
+              : DropdownButton<String>(
+                  // Prevent dropdown errors by first checking that the selected
+                  // device still exists in available inputs.
+                  value: _mediaService.audioInputs.any(
+                    (device) => device.deviceId == selectedAudioDeviceId,
+                  )
+                      ? selectedAudioDeviceId
+                      : null,
+                  items: _mediaService.audioInputs.map((device) {
+                    return DropdownMenuItem<String>(
+                      value: device.deviceId,
+                      child: Text(
+                        device.label!,
+                        style: context.theme.textTheme.titleMedium,
+                      ),
+                    );
+                  }).toList(),
+                  selectedItemBuilder: (BuildContext context) {
+                    return _mediaService.audioInputs.map<Widget>((device) {
+                      return Container(
+                        alignment: Alignment.centerLeft,
+                        constraints: BoxConstraints(
+                          maxWidth: 320,
+                        ),
+                        // 24px for the dropdown arrow
+                        width: MediaQuery.of(context).size.width -
+                            _kTotalDialogContentPadding -
+                            24,
+                        child: Text(
+                          device.label!,
+                          style: context.theme.textTheme.titleMedium,
+                          overflow: TextOverflow.ellipsis,
+                          softWrap: true,
+                        ),
+                      );
+                    }).toList();
+                  },
+                  onChanged: (val) async {
+                    if (val != null) {
+                      setState(() {
+                        selectedAudioDeviceId = val;
+                      });
+                    }
+                  },
```

Comment:
What happens in the case where `val` _is_ `null`? This is the error state mentioned in the comment on line 148, right? I guess the outcome here is "nothing" and I suppose that's the correct outcome. Mostly just commenting because I always wonder when there's neither an "else" nor some other code that occurs after the conditional.

I might rephrase this as

```suggestion
                  onChanged: (val) async {
                    if (val == null) return;
                    setState(() {
                      selectedAudioDeviceId = val;
                    });
                  },
```

It just makes it clearer to the reader that "on `null` we want to do nothing", and most of the time we want to `setState`.